### PR TITLE
feat: Add Harness provider

### PR DESCRIPTION
**Pull request by liran2000**
Add [Harness](https://developer.harness.io/docs/feature-flags) GO Provider.

@davejohnston you are welcome to review from Harness point of view.

@Kavindu-Dodan @toddbaert you are welcome to review from OpenFeature perspective.

### CODE REVIEW

**Review by davejohnston — COMMENTED** [review]
Hi @liran2000 thanks for putting this together.  At a high level it looks good.
I'm just going to take one more look over it and then I'll approve.  

We recently made a change in the goland SDK around initialisation which might help determine when the SDK is ready or not (once its connected and fetched all its config).

File: providers/harness/README.md
Comment (toddbaert) [inline providers/harness/README.md:51]:
```suggestion
context := map[string]interface{}{
```

"target" makes sense, but I think `context` is probably best especially for documentation... unless "target" is a word Harness uses specifically, then I think this is fine.

File: providers/harness/README.md
Comment (toddbaert) [inline providers/harness/README.md:61]: I think you probably want to use the actual SDK here, not the provider, or people might get confused.

File: providers/harness/pkg/provider.go
Comment (toddbaert) [inline providers/harness/pkg/provider.go:19]: @davejohnston is there any risk having multiple clients in a runtime? Do they share any global state? Some SDKs don't play well with that. I suspect not, just pointing it out.

File: providers/harness/pkg/provider.go
Comment (toddbaert) [inline providers/harness/pkg/provider.go:39]: Does harness include any sort of change events? We don't need it for first release, but we could maybe make an issue for it.

File: providers/harness/pkg/provider.go
Comment (toddbaert) [inline providers/harness/pkg/provider.go:100]: You might be able to use generics to reduce some duplication in these functions - but I'm not very experienced with Go generics, so I could be wrong.

File: providers/harness/README.md
Comment (toddbaert) [reply providers/harness/README.md:51]: Nevermind! I can see that "target" is Harness' term.

File: providers/harness/pkg/provider.go
Comment (toddbaert) [inline providers/harness/pkg/provider.go:288]: if `Name` or `Identifier` are special fields for identifying a target/user, we might want to map the `targetingKey` into one of these, and document which it is. We also may want to provide an option for which it's mapped to, if both are used for that purpose.

File: providers/harness/pkg/provider.go
Comment (liran2000) [reply providers/harness/pkg/provider.go:39]: I didn't see events in Harness GO SDK which can be mapped to provider events. If so then it can be address as a follow-up issue yes.

File: providers/harness/pkg/provider.go
Comment (davejohnston) [reply providers/harness/pkg/provider.go]: @toddbaert no we should be good here.  Not all our SDKs behave well but golang is one that we have initialised multiple instances in a single application.

File: providers/harness/pkg/provider.go
Comment (davejohnston) [reply providers/harness/pkg/provider.go:38]: @liran2000 I think this is something we could add into the golang SDK.  Currently we expose a func to tell you if it is initialised.  We debated about exposing a channel to send the event on, but opted not to.  I think what we can do is make it a option to pass into the constructor, where you can pass you're own channel in and get a single when we successfully init

File: providers/harness/pkg/provider.go
Comment (liran2000) [reply providers/harness/pkg/provider.go]: I re-arranged it a bit, such that each evaluation is now more minimal.

**Comment by liran2000** [comment]
> Hi @liran2000 thanks for putting this together. At a high level it looks good. I'm just going to take one more look over it and then I'll approve.

Thanks @davejohnston, I made some changes according to comments, you can have a look and write here if you approve.

File: providers/harness/pkg/provider.go
Comment (liran2000) [reply providers/harness/pkg/provider.go]: @toddbaert trying to understand, why would this can cause having multiple clients in a runtime? can you please elaborate ?

Comment (toddbaert) [reply providers/harness/pkg/provider.go]: Some vendors assume only one SDK client (not openfeature client) exists in the runtime. They share some global state in some cases.

According to @davejohnston this isn't an issue here, at least with Harness' golang SDK.

File: providers/harness/pkg/provider.go
Comment (toddbaert) [reply providers/harness/pkg/provider.go:38]: I think that would be a cool feature. Personally I've always found change events in SDKs useful.

File: providers/harness/README.md
Comment (toddbaert) [inline providers/harness/README.md:24]: Thanks!

**Review by toddbaert — APPROVED** [review]
LGTM, will merge pending @davejohnston 's approval.

**Review by davejohnston — APPROVED** [review]
Looks good, thanks for you’re contribution

**Review by Kavindu-Dodan — APPROVED** [review]

**Pull request merged**
